src/legacy/core_plugins/data/public/index_patterns/index_patterns_service.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { CONTAINS_SPACES } from 'ui/index_patterns/index';
+// @ts-ignore
+import { ILLEGAL_CHARACTERS } from 'ui/index_patterns/index';
+// @ts-ignore
+import { INDEX_PATTERN_ILLEGAL_CHARACTERS } from 'ui/index_patterns/index';
+// @ts-ignore
+import { INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE } from 'ui/index_patterns/index';
+// @ts-ignore
+import { IndexPatternsApiClientProvider } from 'ui/index_patterns/index';
+// @ts-ignore
+import { IndexPatternSelect } from 'ui/index_patterns/index';
+// @ts-ignore
+import { IndexPatternsProvider } from 'ui/index_patterns/index';
+// @ts-ignore
+import { validateIndexPattern } from 'ui/index_patterns/index';
+
+/**
+ * Index Patterns Service
+ *
+ * The `setup` method of this service returns the public contract for
+ * index patterns. Right now these APIs are simply imported from `ui/public`
+ * and re-exported here. Once the index patterns code actually moves to
+ * this plugin, the imports above can simply be updated to point to their
+ * corresponding local directory.
+ */
+export class IndexPatternsService {
+  public setup() {
+    return {
+      IndexPatternsApiClientProvider,
+      IndexPatternsProvider,
+      constants: {
+        ILLEGAL_CHARACTERS,
+        CONTAINS_SPACES,
+        INDEX_PATTERN_ILLEGAL_CHARACTERS,
+        INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE,
+      },
+      ui: {
+        IndexPatternSelect,
+      },
+      validate: {
+        validateIndexPattern,
+      },
+    };
```

Comment:
The shape of what's returned here is completely arbitrary and something we'll need to design on a per-service basis, though I do think we can work to establish some conventions over time.

Like in this case, I've put a React component that index patterns already export under `ui`, and constants under `constants`. But I'm not sure `validate` really needs to be its own thing.

Reply:
right now nothing will use this so i am ok with merging it as it is. we will explore the actual interface as part of deangularization pr (in which we will also update all the consumers of index pattern service to import from this new place)

Reply:
Yep, that makes sense to me. For some future services where we end up exporting stuff like this as an interim step, the way it is structured will become more important. But in the case of index patterns, this was mostly meant to be illustrative.